File: src/TypeScript/TypeScriptLibrary/Library/Measurements/DataConsumerMeasurements.ts
import { FeedbackAliasCollection } from "../FeedbackAliasCollection";
import { FictiveAlias } from "../Fiction/FictiveAlias";
import { FictiveFeedbackAliasCollection } from "../Fiction/FictiveFeedbackAliasCollection";
import { IAlias } from "../Interfaces/IAlias";
import { IDesktop } from "../Interfaces/IDesktop";
import { IFeedbackAlias } from "../Interfaces/IFeedbackAlias";
import { IFeedbackAliasCollection } from "../Interfaces/IFeedbackAliasCollection";
import { IFeedbackAliasCollectionHolder } from "../Interfaces/IFeedbackAliasCollectionHolder";
import { Performer } from "../Performer";
import { DataConsumer } from "./DataConsumer";
import { IMeasurement } from "./Interfaces/IMeasurement";
import { IMeasurements } from "./Interfaces/IMeasurements";

export class DataConsumerMeasurements extends DataConsumer
    implements IMeasurements, IAlias, IFeedbackAliasCollectionHolder
{
    constructor(desktop: IDesktop, name: string) {
        super(desktop, name);
        this.alias = this;
        this.typeName = "DataConsumerMeasurements";
        this.types.push("DataConsumerMeasurements");
        this.types.push("IMeasurements");
        this.types.push("IAlias");
        this.types.push("IExternalAliasDictionary");
    }
    getFeedbackAliasCollection(): IFeedbackAliasCollection {
        return this.feedback;
    }
    addFeedbackAlias(feedbackAlias: IFeedbackAlias): void {
        this.feedbackAliases.push(feedbackAlias);
    }
    updateFeedbackAliasCollection(): void {
        for (let r of this.feedbackAliases) {
            r.setFeedBackAlias();
        }
    }
    getExternalAliasDictionary(): Map<string, string> {
        return this.external;
    }
    getExternalAlias(name: string): string {
        return this.external.get(name) as string;
    }
    addExternalAlias(name: string, value: string): void {
        this.external.set(name, value);
    }

    getAliasValue(name: string)
    {
        return this.aliasValues.get(name);
    }
    protected output: IMeasurement[] = [];

    protected aliasTypes: Map<string, any> = new Map();

    protected aliasValues: Map<string, any> = new Map();

    protected aliasNames: string[] = [];

    protected performer: Performer = new Performer();

    protected variable: any;

    protected alias: IAlias = new FictiveAlias();

    protected external: Map<string, string> = new Map();

    protected feedbackAliases: IFeedbackAlias[] = [];

    protected feedback: IFeedbackAliasCollection = new FictiveFeedbackAliasCollection();

 
    getMeasurementsCount(): number {
        return this.output.length;
    }

    getMeasurement(i: number): IMeasurement {
        return this.output[i];
    }

    addMeasurement(measurement: IMeasurement): void {
        this.output.push(measurement);
    }


    updateMeasurements(): void {
    }

    getAliasType(name: string): any {
        return this.aliasTypes.get(name);
    }
    


    getAliasNames(): string[] {
        return this.aliasNames;
    }

    

    getAliasVаlue(name: string) {
        return this.aliasValues.get(name);
    }

    setAliasValue(name: string, value: any)
    {
        this.performer.setAliasType(name, value, this.aliasTypes, this.aliasNames);
        this.aliasValues.set(name, value);
    }


    setExternalAliases(map: Map<string, string>): void
    {

    }
}

File: Java_Generated/Java/untitled/src/measurements/DataConsumerMeasurements.java
package measurements;

import diagram.interfaces.IDesktop;
import general_service.Entry;
import general_service.FeedbackCollection;
import general_service.interfaces.IAlias;
import general_service.interfaces.IFeedbackCollection;
import general_service.interfaces.IValueSetter;
import measurements.interfaces.IMeasurement;
import measurements.interfaces.IMeasurements;
import measurements.variables.Variable;

import java.util.*;

public class DataConsumerMeasurements  extends  DataConsumer implements IMeasurements, IAlias {



    public DataConsumerMeasurements(String name, IDesktop desktop) {

        super(name, desktop);
    }


    @Override
    public int getMeasurementsCount() {
        return iMeasurements.size();
    }

    @Override
    public IMeasurement getMeasurement(int i) {
        return iMeasurements.get(i);
    }

    @Override
    public void updateMeasurements() {

    }

    @Override
    public void addMeasurement(IMeasurement measurement) {
        iMeasurements.add(measurement);
    }



    public double getInternalTime() {

        return timeMeasurement.getTime();
    }

    List<IMeasurement> iMeasurements = new ArrayList<>();

    @Override
    public String[] getAliasNames() {
        return aliasNames;
    }

    @Override
    public Object getAliasType(String name)
    {
        return map.get(name).getType();
    }

    @Override
    public Object getAliasValue(String name)
    {
        var setter =  map.get(name);
        return setter.getValue();
    }

    @Override
    public void setAliasValue(String name, Object obj) {
       var setter =  map.get(name);
       setter.setValue(obj);
    }

    protected void setMap(java.util.Map<String, Entry<Object, Object>> map)
    {
       aliasNames =  copyMap(map, this.map);
    }

    protected void setFeedback() {
        feedback.setFeedbacks();
    }

    protected void createFeedback() {
        List<Entry<int[], String>> list = new ArrayList<>();
        setFeedback(list);
    }

    protected void setFeedback(List<Entry<int[], String>> list)
    {
        feedback = new FeedbackCollection(this, list);
    }

    protected void addVariableValue(String name, Object type)
    {
    }

    protected void addVariable(Variable variable)
    {
    }


    java.util.Map<String, IValueSetter> map = new HashMap<>();



    protected IFeedbackCollection feedback;

    protected String[] aliasNames = new String[0];

}
